ceph-mon/lib/charms/ceph_mon/v0/ceph_cos_agent.py

```diff
@@ -45,25 +45,41 @@ def __init__(self, charm):
 
     def _on_refresh(self, event):
         """Enable prometheus on relation change"""
-        if not ceph_utils.is_bootstrapped():
-            logger.debug("not bootstrapped, defer _on_refresh: %s", event)
-            event.defer()
+        if not self._charm.unit.is_leader():
+            logger.debug("Not the charm leader, skipping refresh cb.")
```

Comment:
If we return early for non-leaders this means they'll not run the superclass _on_refresh() (which they should)

Reply:
nice catch, thanks for the review :D